docs/examples/pvfleets-qa-pipeline/pvfleets-irradiance-qa.py

```diff
@@ -0,0 +1,375 @@
+                                str(changepoint_amts.index[idx]),
+                                "datetime_end":
+                                    str(changepoint_amts.index[idx + 1]),
+                                "time_shift": change_amt})
+    else:
+        time_shift_list.append({"datetime_start":
+                                str(changepoint_amts.index[idx]),
+                                "datetime_end":
+                                    str(time_shift_series.index.max()),
+                                "time_shift": change_amt})
+
+# Correct any time shifts in the time series
+new_index = pd.Series(time_series.index, index=time_series.index)
+for i in time_shift_list:
+    new_index[(time_series.index >= pd.to_datetime(i['datetime_start'])) &
+              (time_series.index < pd.to_datetime(i['datetime_end']))] = \
+        time_series.index + pd.Timedelta(minutes=i['time_shift'])
+time_series.index = new_index
+
+# Remove duplicated indices and sort the time series (just in case)
+time_series = time_series[~time_series.index.duplicated(
+    keep='first')].sort_index()
+
+# Plot the difference between measured and modeled midday, as well as the
+# CPD-estimated time shift series.
+midday_diff_series.plot()
+time_shift_series.plot()
+plt.title("Midday Difference Time Shift Series")
+plt.show()
+
+# Plot the heatmap of the irradiance time series
+plt.figure()
+# Get time of day from the associated datetime column
+time_of_day = pd.Series(time_series.index.hour +
+                        time_series.index.minute/60,
+                        index=time_series.index)
+# Pivot the dataframe
+dataframe = pd.DataFrame(pd.concat([time_series, time_of_day], axis=1))
+dataframe.columns = ["values", 'time_of_day']
+dataframe = dataframe.dropna()
+dataframe_pivoted = dataframe.pivot_table(index='time_of_day',
+                                          columns=dataframe.index.date,
+                                          values="values")
+plt.pcolormesh(dataframe_pivoted.columns,
+               dataframe_pivoted.index,
+               dataframe_pivoted,
+               shading='auto')
+plt.ylabel('Time of day [0-24]')
+plt.xlabel('Date')
+plt.xticks(rotation=60)
+plt.title('Post-Correction Heatmap, Time of Day')
+plt.colorbar()
+plt.tight_layout()
+plt.show()
+
+# %%
+# Next, we check the time series for any abrupt data shifts. We take the
+# longest continuous part of the time series that is free of data shifts.
+# We use :py:func:`pvanalytics.quality.data_shifts.detect_data_shifts` to
+# detect data shifts in the time series.
+
+# Resample the time series to daily mean
+time_series_daily = time_series.resample('D').mean()
+data_shift_start_date, data_shift_end_date = \
+    ds.get_longest_shift_segment_dates(time_series_daily)
+data_shift_period_length = (data_shift_end_date - data_shift_start_date).days
+
+# Get the number of shift dates
+data_shift_mask = pvanalytics.quality.data_shifts.detect_data_shifts(
```

Comment:
`pvanalytics.quality.data_shifts` was imported as `ds` earlier.

Reply:
addressed in https://github.com/pvlib/pvanalytics/pull/202/commits/cea4fbb93f97e7044f6b50102e1c8bdcf90cd7f4